src/ghg_forcing_for_cmip/download_satellite.py

```diff
@@ -0,0 +1,200 @@
+"""
+Download satellite data
+
+In this module we download the OBS4MIPs data
+used for satellite measurements of CO2 and CH4
+
+"""
+
+import os
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from ecmwf.datastores import Client
+from prefect import flow, task
+
+from ghg_forcing_for_cmip import CONFIG
+from ghg_forcing_for_cmip.utils import (
+    clean_and_save,
+    ensure_trailing_slash,
+    unzip_download,
+)
+from ghg_forcing_for_cmip.validation import EODataSchema
+
+
+@task(
+    name="download_zip_from_cds",
+    description="Download obs4mips zip from Climate Data Store",
+    task_run_name="download_zip_from_cds_{gas}",
+    cache_policy=CONFIG.CACHE_POLICIES,
+)
+def make_api_request(gas: str, save_to_path: str = "data/downloads") -> None:
+    """
+    Request data from Climate Data Store via API
+
+    Parameters
+    ----------
+    gas :
+        target greenhouse gas variable
+        Currently supported gas variables are 'co2' and 'ch4'
+
+    save_to_path :
+        path to save downloaded data
+
+    References
+    ----------
+    Copernicus Climate Change Service, Climate Data Store, (2018):
+    Methane data from 2002 to present derived from satellite observations.
+    Copernicus Climate Change Service (C3S) Climate Data Store (CDS).
+    DOI: 10.24381/cds.b25419f8
+
+    Copernicus Climate Change Service, Climate Data Store, (2018):
+    "Carbon dioxide data from 2002 to present derived from satellite observations".
+    Copernicus Climate Change Service (C3S) Climate Data Store (CDS).
+    DOI: 10.24381/cds.f74805c8
+
+    Notes
+    -----
+    A login and further pre-setup is required to be
+    able to download the data from the API.
+    Please refer to the following link for further details:
+    https://cds.climate.copernicus.eu/how-to-api
+    """
+    # TODO: Is there a way to check whether the /.cdsapirc file is
+    #  somewhere on the computer?
+
+    if gas == "co2":
+        dataset = "satellite-carbon-dioxide"
+    elif gas == "ch4":
+        dataset = "satellite-methane"
+    else:
+        raise ValueError(f"gas must be 'ch4', 'co2'. Got {gas=}")  # noqa: TRY003
+
+    # create directory if it doesn't exist
+    os.makedirs(save_to_path, exist_ok=True)
+
+    request = {
+        "processing_level": ["level_3"],
+        "variable": f"x{gas}",
+        "sensor_and_algorithm": "merged_obs4mips",
+        "version": ["4_5"],
+    }
+    # setup saving location
+    target = save_to_path + f"/obs4mips_x{gas}.zip"
```

Comment:
Using a `Path` for `save_to_path` means this can become `save_to_path / f"obs4mips_x{gas}.zip"` which is maybe slightly easier to read (small thing, but can help for cross-platform compatibility as I think hard-coding `/` might not work on windows?)